### PR TITLE
Added catkin support

### PR DESCRIPTION
**Pull request by 4c3y**
Added support to build the package with `catkin build`

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -29,16 +29,30 @@ find_package(catkin REQUIRED COMPONENTS roscpp)
...
 if (catkin_FOUND)
+        set(catkin_INCLUDE_DIRS ${catkin_INCLUDE_DIRS} ${LPP_INCLUDE_DIRS})
         include_directories(${catkin_INCLUDE_DIRS})
         target_link_libraries(lpp_demo ${catkin_LIBRARIES})
+
+        install(TARGETS lpp
+                ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+                LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+                RUNTIME DESTINATION ${CATKIN_GLOBAL_BIN_DESTINATION}
+                )
```
Comment (michaelpantic) [inline CMakeLists.txt:55]: does this work if we don't use install spaces?

File: package.xml [hunk 2]
```diff
@@ -6,4 +6,5 @@
   <maintainer email="69460051+4c3y@users.noreply.github.com">4c3y</maintainer>
   <license>BSD</license>
   <buildtool_depend>catkin</buildtool_depend>
+  <depend>roscpp</depend>
```
Comment (michaelpantic) [inline package.xml:9]: in the off case, where catkin is used but ros isnt, this will fail.

but I guess at the moment we seldomly have this case.

**Review by michaelpantic — APPROVED** [review]
LGTM! 

2 Questions/comments but otherwise good to merge

File: CMakeLists.txt (hunk 1 above)
Comment (4c3y) [reply CMakeLists.txt:55]: Yes because all env variables are set by default according to the catkin cmake documentation

File: package.xml (hunk 2 above)
Comment (4c3y) [reply package.xml:9]: I'll leave it for now and fix it if necessary.

**Pull request merged**